packages/devextreme/js/__internal/ui/m_tag_box.ts

```diff
@@ -976,7 +984,9 @@ const TagBox = (SelectBox as any).inherit({
 
     if (needFilterPlainItems) {
       const plainItems = this._getPlainItems();
-      selectedItems = this._filterSelectedItems(plainItems, values);
+      const filteredItems = this._filterSelectedItems(plainItems, values);
+
+      selectedItems = this.option('hideSelectedItems') ? this.option('selectedItems').concat(filteredItems) : filteredItems;
```

Comment:
There are a lot of changes, im not quite follow them.
 As far as i understand, when 'hideSelectedItems' option is true, we hide selected items from the list and they appear as tagBox values. At some point, when item in the list is selected, but not yet reflected in tagbox, to get selectedItems we actually have to combine 2 arrays. There is a code inside _getFilteredItems function to manage selected items

selectedItems = listSelectedItems || this.option('selectedItems');

In case when 'hideSelectedItems' is true we can do something like this
selectedItems = (listSelectedItems || []).concat(this.option('selectedItems') || []);

I believe this is the only change that is needed

Reply:
[d66f35b](https://github.com/DevExpress/DevExtreme/pull/27521/commits/d66f35bc7c4fa23dd30f6d01d8a41748040bce4e)